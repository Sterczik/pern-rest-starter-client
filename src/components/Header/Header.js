import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { authActions } from '../../containers/App/auth/actions';

export const Header = ({ logout }) => (
    <header>
        <Link to="/dashboard">
            <h1>Hi</h1>
        </Link>
        <button onClick={logout}>Logout</button>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    logout: () => dispatch(authActions.logout())
});

export default connect(undefined, mapDispatchToProps)(Header);