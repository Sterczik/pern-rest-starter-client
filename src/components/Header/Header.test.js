import React from 'react';
import { shallow } from 'enzyme';
import { Header } from './Header';

let wrapper;
let logout;

beforeEach(() => {
  logout = jest.fn();
  wrapper = shallow(<Header
    logout={logout}
  />);
});

test('should render Header correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handle logout', () => {
  wrapper.find('button').simulate('click');
  expect(logout).toHaveBeenCalled();
});
