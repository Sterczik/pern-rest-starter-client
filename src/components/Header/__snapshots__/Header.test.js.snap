// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<div
  className="header"
>
  <WithStyles(AppBar)
    position="static"
  >
    <WithStyles(Toolbar)>
      <WithStyles(Typography)
        className="header__typo"
        color="inherit"
        variant="headline"
      >
        <Link
          className="header__link"
          replace={false}
          to="/todos"
        >
          Starter
        </Link>
      </WithStyles(Typography)>
      <WithStyles(IconButton)
        aria-haspopup="true"
        color="inherit"
        onClick={[Function]}
      >
        <pure(AccountCircleIcon) />
      </WithStyles(IconButton)>
      <WithStyles(Menu)
        anchorEl={null}
        id="simple-menu"
        onClose={[Function]}
        open={false}
      >
        <WithStyles(MenuItem)
          className="header__anchor"
          component={[Function]}
          to="/todos"
        >
          Todos
        </WithStyles(MenuItem)>
        <WithStyles(MenuItem)
          className="header__anchor"
          component={[Function]}
          to="/my-account"
        >
          My account
        </WithStyles(MenuItem)>
        <WithStyles(MenuItem)
          className="header__anchor"
          onClick={[Function]}
        >
          Logout
        </WithStyles(MenuItem)>
      </WithStyles(Menu)>
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>
</div>
`;
